File: ruby/lib/steam/community/tf2/tf2_item.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2010-2011, Sebastian Staudt

require 'steam/community/game_item'

# Represents a Team Fortress 2 item#
#
# @author Sebastian Staudt
class TF2Item

  include GameItem

  CLASSES = [ :scout, :sniper, :soldier, :demoman, :medic, :heavy, :pyro, :spy ]

  # Creates a new instance of a TF2 item with the given data
  #
  # @param [TF2Inventory] inventory The inventory this item is contained in
  # @param [Hash<Symbol, Object>] item_data The data specifying this item
  # @throws WebApiException on Web API errors
  def initialize(inventory, item_data)
    super

    @equipped = {}
    CLASSES.each_index do |class_id|
      @equipped[CLASSES[class_id]] = (item_data[:inventory] & (1 << 16 + class_id) != 0)
    end
  end

  # Returns the class symbols for each class this player has equipped this item
  #
  # @return [Array<String>] The names of the classes this player has equipped
  #         this item
  def classes_equipped?
    @equipped.reject { |class_id, equipped| !equipped }
  end

  # Returns whether this item is equipped by this player at all
  #
  # @return `true` if the player has equipped this item at all
  def equipped?
    @equipped.has_value? true
  end

end

File: php/lib/steam/community/tf2/TF2Item.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2010-2011, Sebastian Staudt
 *
 * @author     Sebastian Staudt
 * @license    http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package    Steam Condenser (PHP)
 * @subpackage Steam Community
 */

require_once STEAM_CONDENSER_PATH . 'steam/community/GameItem.php';
require_once STEAM_CONDENSER_PATH . 'steam/community/tf2/TF2Inventory.php';

/**
 * Represents a Team Fortress 2 item
 *
 * @package    Steam Condenser (PHP)
 * @subpackage Steam Community
 */
class TF2Item extends GameItem {

    /**
     * @var array
     */
    private static $CLASSES = array('scout', 'sniper', 'soldier', 'demoman',
                                    'medic', 'heavy', 'pyro', 'spy');

    /**
     * @var array
     */
    private $equipped;

    /**
     * Creates a new instance of a TF2Item with the given data
     */
    public function __construct(TF2Inventory $inventory, $itemData) {
        parent::__construct($inventory, $itemData);

        $this->equipped = array();
        foreach(self::$CLASSES as $classId => $className) {
            $this->equipped[$className] = ($itemData->inventory & (1 << 16 + $classId) != 0);
        }
    }

    /**
     * Returns the class symbols for each class this player has equipped this item
     */
    public function getClassesEquipped() {
        $classesEquipped = array();
        foreach($this->equipped as $classId => $equipped) {
            if($equipped) {
                $classesEquipped[] = $classId;
            }
        }

        return $classesEquipped;
    }

    /**
     * Returns whether this item is equipped by this player at all
     */
    public function isEquipped() {
        return in_array(true, $this->equipped);
    }

}
?>
